### PR TITLE
dot graph

### PR DESCRIPTION
**Pull request by sinelaw**
Add --graph FILE and --dry-run to allow producing a dot file of dependencies with/without actually running the linker command

### CODE REVIEW

**Comment by ElastiLotem** [comment]
Instead of dry run, you can already prepend `echo` to your command. Perhaps just have the ld command be `<|>` with the `--graph` option (exclusively) and then generate a graph OR run the ld command?

**Comment by sinelaw** [comment]
Regardin dry-run: it's convenient this way because it prints out the exact command it would run
As for --graph: users may want to produce the graph as they build (e.g. add --graph optionally via a build system configure flag).

**Comment by ElastiLotem** [comment]
Regarding --dry-run: So does prepending "echo"? :)

--graph: Yeah, not exclusive is perhaps good.

**Comment by sinelaw** [comment]
Yes, but it involves changing a specific part of the command line. The advantage here is that you have a separate flag you can add arbitrarily in the options list to disable all side effects. Also it's more future proof, in case we want to add more actions (e.g. multiple commands), --dry-run will always disable all of them

**Comment by sinelaw** [comment]
Merge meeeeeeeee

File: Main.hs [hunk 1]
```diff
@@ -54,14 +58,27 @@ getOpts =
            metavar "ld-command")
       <*> some (argument bytestr (metavar "opaths" <> help "At least one root .o path"))
       <*> switch (long "verbose" <> short 'v' <> help "Verbose mode")
+      <*> switch (long "dry-run" <> short 'd' <> help "Dry run (won't execute the command)")
+      <*> optional (option bytestr (metavar "DOTFILE" <> long "graph" <> short 'g' <> help "Generate dependency graph (in GraphViz dot format)"))
+
+-- TODO: add escaping, etc.
+quote :: ByteString -> ByteString
+quote x = "\"" <> x <> "\""
```
Comment (ElastiLotem) [inline Main.hs]: Maybe go thru 'show' ordinarily?

Comment (sinelaw) [inline Main.hs]: Updated

**Comment by ElastiLotem** [comment]
Thanks!!

**Pull request merged**